src/sha1.cpp

```diff
@@ -199,15 +201,15 @@ cvk_sha1_hash cvk_sha1(const uint8_t* data, uint32_t length) {
     // Process data in 512-bit blocks.
     while (length >= 64) {
         // Copy block to local buffer and process it.
-        memcpy(block, data, 64);
+        memcpy(block, data_i8, 64);
         sha1_process_block(H, block);
 
         length -= 64;
-        data += 64;
+        data_i8 += 64;
```

Comment:
I hadn't spotted the pointer arithmetic here. Probably still better to have the cast in the implementation.